File: src/dataset/python/anagram_exercism-ThomasZumsteg/gemini/GeminiFlash_anagram_v3.py
"""Finds anagrams of a word in a list of words"""

def detect_anagrams(anagram, word_list):
    """Finds any words in [word_list] that are anagrams of [anagram]"""
    anagram_lower = anagram.lower()
    letters = sorted(anagram_lower)
    anagram_len = len(anagram)

    return [word for word in word_list
            if len(word) == anagram_len and word.lower() != anagram_lower and sorted(word.lower()) == letters]

File: src/dataset/Java/anagram_exercism-ThomasZumsteg/gemini/GeminiFlash_Anagram_v3.java
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class Anagram {
    private final String anagramLower;
    private final char[] anagramLetters;

    public Anagram(String anagram) {
        this.anagramLower = anagram.toLowerCase();
        this.anagramLetters = this.anagramLower.toCharArray();
        Arrays.sort(this.anagramLetters);
    }

    public List<String> match(List<String> words) {
        List<String> anagrams = new ArrayList<>();
        for (String word : words) {
            if (isAnagram(word)) {
                anagrams.add(word);
            }
        }
        return anagrams;
    }

    private boolean isAnagram(String word) {
        String wordLower = word.toLowerCase();

        if (anagramLower.equals(wordLower)) {
            return false;
        }

        if (wordLower.length() != anagramLower.length()) {
            return false;
        }

        char[] wordLetters = wordLower.toCharArray();
        Arrays.sort(wordLetters);

        return Arrays.equals(anagramLetters, wordLetters);
    }
}